test/admin-hpa/features/step_definitions/data_sync_steps.rb

```diff
@@ -4,8 +4,11 @@
 end
 
 When(/^the data sync function has run$/) do
-  response = FunctionsHelper.resync_all(true)
-  expect(response.code).to eql 202
+  if @check_code
+    (wait_until(ENV['WAIT_TIME'].to_i, 2) {SqlDbHelper.get_pupil_check_metadata(@check_code)['checkStatus_id'] == 2}) if @no_answers.nil?
```

Comment:
This is a brittle check.  The status code is not guaranteed to be any number.   Please check against the checkStatus.code lookup table for 'CMP'.